### PR TITLE
Mobile: Try running all UI tests on mobile PRs

### PR DESCRIPTION
**Pull request by guarani**
## Description

An experimental PR to see what it would look like if we were to run _all_ React Native UI tests on mobile PRs (i.e. those with the `Mobile App - i.e. Android or iOS` label).

## How has this been tested?

```
@mchowning 
```

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: .github/workflows/rnmobile-ios-runner.yml [hunk 1]
```diff
@@ -2,16 +2,18 @@ name: React Native E2E Tests (iOS)
 
 on:
     pull_request:
+        type: [labeled, opened, synchronize, reopened]
```
Comment (guarani) [inline .github/workflows/rnmobile-ios-runner.yml:5]: `labeled` ensures the workflow is only executed on labelled PRs, the other three types of PRs I've added appear to be defaults which I'm not sure are needed explicitly now that I'm customizing this: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues

File: .github/workflows/rnmobile-ios-runner.yml [hunk 2]
```diff
@@ -2,16 +2,18 @@ name: React Native E2E Tests (iOS)
...
     push:
         branches: [trunk]
 
 jobs:
     test:
+        if: ${{ github.event.label.name == 'Mobile App - i.e. Android or iOS' }}
```
Comment (guarani) [inline .github/workflows/rnmobile-ios-runner.yml:11]: Ideally we wouldn't have to duplicate the workflow or large parts of it, but not sure how to do that so for now I'm just making the test run only on mobile PRs (this is just an experiment, the plan is not to merge this change as-is).

Comment (guarani) [reply .github/workflows/rnmobile-ios-runner.yml:11]: To clarify, the idea I have is to run only a small UI test suite on web PRs and run all UI tests on mobile PRs. So hopefully I'll find a way to do that with a conditional if-else that doesn't duplicate lots of code. For now I've made these mobile UI tests run _only_ on mobile PRs.

**Comment by youknowriad** [comment]
Doing some cleanup to old PRs. This draft PR seem staled. I'm closing this PR. Please let me know if I missed anything.

**Pull request closed without merging**